src/screens/main/Main.tsx

```diff
@@ -34,7 +38,15 @@ export const Main: React.FC = () => {
         stickyHeaderIndices={[0]}
         numColumns={2}
         data={products}
-        renderItem={({item}) => <ProductCard key={item.id} product={item} />}
+        renderItem={({item}) => (
+          <ProductCard
+            key={item.id}
+            product={item}
+            onPress={product =>
+              navigation.navigate('ProductDetails', {productId: product.id})
```

Comment:
Cause you use only `id` here it will be better to pass just `id`:
```js
            onPress={productId =>
              navigation.navigate('ProductDetails', {productId})
            }
```